libflux/flux-core/src/semantic/types.rs

```diff
@@ -374,6 +391,7 @@ impl MonoType {
         }
     }
 
+    #[allow(missing_docs)]
     pub fn constrain(self, with: Kind, cons: &mut TvarKinds) -> Result<Substitution, Error> {
```

Comment:
Constrain validates that the current type meets the constraints of the specified kind.